tests/multiprocess_gpu_test.py

```diff
@@ -38,8 +46,72 @@
 except ImportError:
   pytest = None
 
+config.update("experimental_xmap_spmd_lowering", True)
 config.parse_flags_with_absl()
 
+X_SIZE = 4
+Y_SIZE = 2
+Z_SIZE = 2
+ALL_AXES = ("x", "y", "z")
+
+def jax_multiprocess_init():
+  # Hookup the ENV vars expected to be set already in the SLURM environment
+  coordinator_address = os.environ.get("SLURM_STEP_NODELIST", None)
+  if coordinator_address is not None and '[' in coordinator_address:
+    coordinator_address = coordinator_address.split('[')[0] + \
+                          coordinator_address.split('[')[1].split(',')[0]
+  num_tasks = os.environ.get("SLURM_NPROCS", None)
+  taskid = os.environ.get("SLURM_PROCID", None)
+  localid = os.environ.get("SLURM_LOCALID", None)
+
+  # fixing port since it needs to be the same for all the processes
+  port = "54321"
+
+  print(f"coord addr:port : {coordinator_address}:{port}\nTotal tasks: "
+        f"{num_tasks}\ntask id: {taskid}\nlocal id: {localid}")
+
+  assert (coordinator_address is None or num_tasks is None or
+          taskid is None) is False
+
+  jax.config.update("jax_cuda_visible_devices", localid)
+
+  jax.distributed.initialize(
```

Comment:
Now that Nicholas's PR is submitted (or will be soon), do you need all this setup? Can you just use `jax.distributed.initialize()`?

Reply:
I think that's a good idea, will use that